src/time_evolution/ssesolve.jl

```diff
@@ -0,0 +1,409 @@
+- `tlist::AbstractVector`: List of times at which to save the state of the system.
+- `sc_ops::Union{Nothing,AbstractVector}=nothing`: List of stochastic collapse operators ``\{\hat{C}_n\}_n``.
+- `alg::StochasticDiffEq.StochasticDiffEqAlgorithm`: Algorithm to use for the time evolution.
+- `e_ops::Union{Nothing,AbstractVector}`: List of operators for which to calculate expectation values.
+- `H_t::Union{Nothing,Function,TimeDependentOperatorSum}`: Time-dependent part of the Hamiltonian.
+- `params::NamedTuple`: Dictionary of parameters to pass to the solver.
+- `seeds::Union{Nothing, Vector{Int}}`: List of seeds for the random number generator. Length must be equal to the number of trajectories provided.
+- `n_traj::Int`: Number of trajectories to use.
+- `ensemble_method`: Ensemble method to use.
+- `prob_func::Function`: Function to use for generating the SDEProblem.
+- `output_func::Function`: Function to use for generating the output of a single trajectory.
+- `kwargs...`: Additional keyword arguments to pass to the solver.
+
+# Notes
+
+- `ensemble_method` can be one of `EnsembleThreads()`, `EnsembleSerial()`, `EnsembleDistributed()`
+- The states will be saved depend on the keyword argument `saveat` in `kwargs`.
+- If `e_ops` is specified, the default value of `saveat=[tlist[end]]` (only save the final state), otherwise, `saveat=tlist` (saving the states corresponding to `tlist`). You can also specify `e_ops` and `saveat` separately.
+- The default tolerances in `kwargs` are given as `reltol=1e-6` and `abstol=1e-8`.
+- For more details about `alg` and extra `kwargs`, please refer to [`DifferentialEquations.jl`](https://diffeq.sciml.ai/stable/)
+
+# Returns
+
+- `sol::TimeEvolutionSSESol`: The solution of the time evolution. See also [`TimeEvolutionSSESol`](@ref)
+"""
+function ssesolve(
+    H::QuantumObject{MT1,OperatorQuantumObject},
+    ψ0::QuantumObject{<:AbstractArray{T2},KetQuantumObject},
+    tlist::AbstractVector,
+    sc_ops::Union{Nothing,AbstractVector} = nothing;
+    alg::StochasticDiffEq.StochasticDiffEqAlgorithm = SRA1(),
+    e_ops::Union{Nothing,AbstractVector} = nothing,
+    H_t::Union{Nothing,Function,TimeDependentOperatorSum} = nothing,
+    params::NamedTuple = NamedTuple(),
+    n_traj::Int = 1,
+    ensemble_method = EnsembleThreads(),
+    prob_func::Function = _ssesolve_prob_func,
+    output_func::Function = _ssesolve_output_func,
+    kwargs...,
+) where {MT1<:AbstractMatrix,T2}
+    ens_prob = ssesolveEnsembleProblem(
+        H,
+        ψ0,
+        tlist,
+        sc_ops;
+        alg = alg,
+        e_ops = e_ops,
+        H_t = H_t,
+        params = params,
+        prob_func = prob_func,
+        output_func = output_func,
+        kwargs...,
+    )
+
+    return ssesolve(ens_prob; alg = alg, n_traj = n_traj, ensemble_method = ensemble_method)
+end
+
+function ssesolve(
+    ens_prob::EnsembleProblem;
+    alg::StochasticDiffEq.StochasticDiffEqAlgorithm = SRA1(),
```

Comment:
if we import `StochasticDiffEq` with keyword `import` in `src/QuantumToolbox.jl`, I think we can just put
```julia
alg::StochasticDiffEqAlgorithm = SRA1(),
```
here